### PR TITLE
Add com.jfrog.includeAllBuildFiles flag

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](../CONTRIBUTING.md#tests) passed. If this feature is not already covered by the tests, I added new
  tests.

-----

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ purposes and applications as well, by applying it in your build.gradle file.
 
 Inject the plugin using the [init.gradle](./init.gradle) initialization script, and run *generateDepTrees* in a
 directory containing a build.gradle file. The plugin will generate a dependency tree for each subproject that does not
-contain a build.gradle file.
+contain a build.gradle file. To generate a dependency tree for each project that contains a build file, set the `-Dcom.jfrog.includeAllBuildFile` flag to `true`.
```
Comment (yahavi) [inline README.md:14]:
```suggestion
contain a build.gradle file. To generate a dependency tree for each subproject that contains a Gradle build file, set the `-Dcom.jfrog.includeAllBuildFile` flag to `true`.
```

File: src/main/java/com/jfrog/tasks/GenerateDepTrees.java [hunk 2]
```diff
@@ -127,9 +127,10 @@ public TaskDependency getFinalizedBy() {
     private List<Project> getRelatedProjects() {
         List<Project> relatedProjects = new ArrayList<>();
         relatedProjects.add(getProject());
+        boolean includeAllBuildFiles = Boolean.parseBoolean(System.getProperty(INCLUDE_ALL_BUILD_FILES));
 
-        for (Project project : (Set<Project>) getProject().getSubprojects()) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/tasks/GenerateDepTrees.java:131]: I'm not sure this will work on Java 8

**Review by yahavi — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (omerzi) [reply README.md:14]: done

File: src/main/java/com/jfrog/tasks/GenerateDepTrees.java (hunk 2 above)
Comment (omerzi) [reply src/main/java/com/jfrog/tasks/GenerateDepTrees.java:131]: done

File: src/functionalTest/java/com/jfrog/tasks/IncludeAllBuildFilesTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import static com.jfrog.tasks.Consts.MULTI;
+import static com.jfrog.tasks.Consts.TEST_DIR;
+import static com.jfrog.tasks.Utils.*;
+import static org.testng.Assert.*;
+
+/**
+ * Functional tests for the project under resources/multi/
+ * This project contain subprojects in a multiple build.gradle files.
+ *
+ * @author yahavi
```
Comment (yahavi) [inline src/functionalTest/java/com/jfrog/tasks/IncludeAllBuildFilesTest.java:26]: Please update the description

**Review by yahavi — CHANGES_REQUESTED** [review]

**Pull request merged**